nifi-commons/nifi-expression-language/src/main/java/org/apache/nifi/attribute/expression/language/evaluation/functions/IsJsonEvaluator.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.attribute.expression.language.evaluation.functions;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.nifi.attribute.expression.language.EvaluationContext;
+import org.apache.nifi.attribute.expression.language.evaluation.BooleanEvaluator;
+import org.apache.nifi.attribute.expression.language.evaluation.BooleanQueryResult;
+import org.apache.nifi.attribute.expression.language.evaluation.Evaluator;
+import org.apache.nifi.attribute.expression.language.evaluation.QueryResult;
+
+import java.io.IOException;
+
+public class IsJsonEvaluator extends BooleanEvaluator {
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private final Evaluator<String> subject;
+
+    public IsJsonEvaluator(Evaluator<String> subject) {
+        this.subject = subject;
+    }
+
+    @Override
+    public QueryResult<Boolean> evaluate(EvaluationContext evaluationContext) {
+        final String subjectValue = subject.evaluate(evaluationContext).getValue();
```

Comment:
You should consider checking the first and last characters for opening and closing JSON blocks (array and object open/close) so that you can short circuit the evaluation of strings which don't even superficially resemble JSON. While literals can technically be valid JSON, I'm not sure if the trade off is worth it.

Reply:
I agree with @MikeThomsen. Although literals are technically valid JSON, I can see it being confusing.

Perhaps naming the function more specifically, like `isJsonNode()` would be helpful. Naming it `isJsonObject()` could also work, but that raises the question of whether there should be one called `isJsonArray()`.

Reply:
@exceptionfactory  Correct me if I am wrong but it seems JsonNode is associated more with implementation as that is what Jackson names its object. To me at least,  isJsonNode would indicate how this is implemented and is not as important to the end user who might not even know what JsonNode is. Hence if you don't mind, I will leave the method as isJson, implement the change suggested by @MikeThomsen and document the purpose of the function indicating what it can and what it cannot do.

Reply:
That's a good point @dan-s1, `JsonNode` is more closely associated with Jackson, so I agree it is not the best name. Keeping it as `isJson` sounds good in absence of other suggestions.

Reply:
@MikeThomsen I made the changes as you requested. You were absolutely correct as with the requested checks it is easier to test and document.
@exceptionfactory  I made the requested changes but I first updated my branch to be updated with main. I am not sure though I was supposed to do that. Please advise.